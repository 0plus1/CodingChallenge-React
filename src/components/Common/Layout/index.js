/* React */
import React from 'react';


function Index(components) {
  const { Content } = components;

  // eslint-disable-next-line react/prefer-stateless-function
  class Component extends React.Component {
    render() {
      return (
        // eslint-disable-next-line react/jsx-filename-extension
        <div className="console">
          <Content {...this.props} />
        </div>
      );
    }
  }

  Component.displayName = 'Layout';

  return Component;
}

export default Index;
