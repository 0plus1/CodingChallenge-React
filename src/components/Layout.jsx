import React from 'react';
import PropTypes from 'prop-types';
import { Container, AppBar, Toolbar, Typography, Button } from '@material-ui/core';
import styled from 'styled-components';
import StyledLink from './StyledLink';

const StyledContainer = styled(Container)`
  margin-top: 2em;
`;

const Layout = WrappedComponent => class extends React.PureComponent {
  render() {
    return (
      <React.Fragment>
        <AppBar position="static">
          <Toolbar>
            <Typography variant="h6">
              Book Finder
            </Typography>
            <Button color="inherit"><StyledLink to={'/'}>Home</StyledLink></Button>
          </Toolbar>
        </AppBar>
        <StyledContainer id="body" maxWidth="lg">
          <WrappedComponent {...this.props} />
        </StyledContainer>
      </React.Fragment>
    );
  }
};

Layout.proptypes = {
  children: PropTypes.required,
};

export default Layout;
