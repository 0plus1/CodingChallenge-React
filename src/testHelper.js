import React from 'react';
import MemoryRouter from 'react-router/MemoryRouter';
import { mount, render, shallow } from 'enzyme';

const mountWithRouter = node => mount(<MemoryRouter>{node}</MemoryRouter>);
const renderWithRouter = node => render(<MemoryRouter>{node}</MemoryRouter>);
const shallowWithRouter = node => shallow(<MemoryRouter>{node}</MemoryRouter>);

export {
  mountWithRouter,
  renderWithRouter,
  shallowWithRouter,
};
