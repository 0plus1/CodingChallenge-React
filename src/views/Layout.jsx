import React from 'react';
import PropTypes from 'prop-types';
import { Container } from '@material-ui/core';

const Layout = WrappedComponent => class extends React.PureComponent {
  render() {
    return (
        <Container id="body" maxWidth="lg">
          <WrappedComponent {...this.props} />
        </Container>
    );
  }
};

Layout.proptypes = {
  children: PropTypes.required,
};

export default Layout;
