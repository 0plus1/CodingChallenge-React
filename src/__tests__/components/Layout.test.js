import React from 'react';
import Layout from '../../components/Layout';
import { mountWithRouter, renderWithRouter, shallowWithRouter } from '../../testHelper';

describe('<Layout />', () => {
  describe('renders', () => {
    it('without crashing', () => {
      shallowWithRouter(<Layout />);
      expect(<Layout />).toMatchSnapshot();
    });

    it('with child components', () => {
      const child = () => <div className="child">test</div>;
      const TestHOC = Layout(child);
      const wrapper = mountWithRouter(<TestHOC />);
      expect(wrapper.find(child)).toHaveLength(1);
      wrapper.unmount(); // enzyme dom cleanup

      const renderer = renderWithRouter(<TestHOC />);
      expect(renderer.find('div.child')).toHaveLength(1);
    });
  });
});
