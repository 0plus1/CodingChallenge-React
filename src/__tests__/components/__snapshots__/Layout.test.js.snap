// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders without crashing 1`] = `<Layout />`;
